### PR TITLE
Migrated ocall test to use oeedger8r.

### PR DESCRIPTION
**Pull request by EMumau**
Updated the ocall test to use an EDL file and the oeedger8r.
In order to do this, I also created the ocall_callback test because I was unable to get the callback test working with the oeedger8r generated code.  I isolated the callback test case and created a separate binary for it.

### CODE REVIEW

**Comment by ihaller** [comment]
Out of curiosity, why are the new tests not additive to the existing ones? My understanding was that raw ECALLs and OCALLs continue to be supported and thus there is value in maintaining their test suite.

**Comment by paulcallen** [comment]
yes, this is one of the few sets of tests that probably need to keep a full set of tests doing ecalls and ocalls the old fashioned way.
We can have new tests that do edger8r versions and the old versions left as is to test the legacy way of doing calls.

File: tests/ocall/enc/enc.cpp [hunk 1]
```diff
@@ -3,31 +3,28 @@
...
-OE_ECALL void Test4(void* args)
+void test4(void)
 {
     unsigned char buf[32];
 
     /* Call into host with enclave memory */
     oe_memset(buf, 0xAA, sizeof(buf));
 
-    if (oe_call_host("Func2", buf) != OE_OK)
+    if (func2() != OE_OK)
```
Comment (anakrish) [inline tests/ocall/enc/enc.cpp]: The earlier version passed enclave buffer (buf) to host. 
The new version ought to do that too.

File: tests/ocall/ocall.edl [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        public uint64_t test_my_ocall();
+
+        public void test_ocall_edge_cases();
+
+        public void test_reentrancy_ecall ();
+    };
+
+    untrusted {
+        void func1 ();
```
Comment (anakrish) [inline tests/ocall/ocall.edl]: Can you prefix the host functions with host_ to make reading easier?
For ecalls always the first argument is an enclave and it is obvious that we are making an ecall.
However for ocalls, there is no such argument that makes it clear that we are doing an ocall.

**Comment by EMumau** [comment]
This pull request is no longer valid.  This work has been completed in a different pull request.  I am closing it out.

**Pull request closed without merging**